src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+                return Resolv::DNS.new.getresource("_kerberos._tcp.dc._msdcs.#{ad_domain_name}", Resolv::DNS::Resource::IN::SRV).target.to_s
+            rescue Resolv::ResolvError
+                return ''
+            end
+        end
+
+        # Return list of package names that should be installed to satisfy all configuration requirements.
+        def calc_pkg_deps
+            pkgs = []
+            if @sssd_enabled || @sssd_pam || @sssd_nss.any?
+                pkgs += ['sssd']
+                # Only install the required provider packages. By convention, they are named 'sssd-*'.
+                domain_providers = ['ad', 'ldap', 'ipa', 'proxy', 'krb5']
+                @sssd_conf.each { |sect_name, conf|
+                    id_provider = conf['id_provider']
+                    auth_provider = conf['auth_provider']
+                    if domain_providers.include?(id_provider)
+                        pkgs += ['sssd-' + id_provider]
+                    end
+                    if domain_providers.include?(id_provider)
+                        pkgs += ['sssd-' + id_provider]
+                    end
+                }
+            end
+            if @ldap_pam
+                pkgs += ['pam_ldap']
+            end
+            if @krb_pam
+                pkgs += ['pam_krb5', 'krb5', 'krb5-client']
+            end
+            if @ldap_nss
+                pkgs += ['nss_ldap']
+                if @ldap_nss.include?('automount')
+                    pkgs += ['openldap2-client'] # provides /etc/openldap/ldap.conf
+                end
+            end
+            if @autofs_enabled || @sssd_nss.include?('automount') || @ldap_nss.include?('automount')
+                pkgs += ['autofs']
+            end
+            if @nscd_enabled
+                pkgs += ['nscd']
+            end
+            return pkgs
+        end
+
+        # Summarise the authentication configurations in a human-readable, single line of text.
+        def summary_text
+            # Figure out how authentication works on this computer
+            auth_doms_caption = ''
+            if !@sssd_enabled && @ldap_nss.empty? && !@ldap_pam && !@krb_pam
+                # Local only
+                auth_doms_caption = _('Only use local authentication')
+            elsif @sssd_enabled && (@sssd_pam || @sssd_nss.any?)
+                # SSSD is configured
+                if Service.Active('sssd')
+                    auth_doms_caption = @sssd_conf['sssd']['domains'].join(', ')
+                else
+                    auth_doms_caption = _('Configured but not activated')
+                end
+            else
+                # LDAP and/or Kerberos is configured
+                if @ldap_nss.any? || @ldap_pam
+                    if @ldap_conf['base'].to_s == ''
+                        auth_doms_caption = _('LDAP is enabled but the setup is incomplete')
+                    else
+                        auth_doms_caption = _('via LDAP on ') + @ldap_conf['base']
```

Comment:
Translatable texts must not be constructed with string concatenation, because that makes them untranslatable into languages with different word order: for example in Japanese , "via LDAP on example.base" is "LDAP-on example.base-via" (and similarly in Finnish!)

So this particular line should become
```rb
# TRANSLATORS: %s is a LDAP base name
auth_doms_caption = _('via LDAP on %s') % @ldap_conf['base']

```